File: v1/Demo/source/demo_get_velocity_limited.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
    usleep(1000000);

    vector<float> velocity_limited = api.get_velocity_limited();
    if (velocity_limited.size() != 6)
    {
        Logger::get_instance()->print_trace_error("get velocity_limited failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get velocity_limited success : ";
    for (int i = 0; i < velocity_limited.size(); ++i) {
        std::cout << velocity_limited[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}

File: v1/Demo/demo_get_velocity_limited.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *


def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    velocity_limited = api.get_velocity_limited()

    if len(velocity_limited) != 6:
        logger.print_trace_error("velocity_limited failed")
    else:
        logger.print_trace("velocity_limited success", velocity_limited)

if __name__ == "__main__":
    main()


